### PR TITLE
add make apply-dependencies and update developer guide

### PR DESCRIPTION
**Pull request by oliviassss**
*Issue #, if available:*

*Description of changes:*

This PR adds `apply-dependencies` to Makefile to install the cert manager and certificate. Also updates the developer's guide for this change. Tested in dev setup.
 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -89,7 +89,21 @@ __check_var = \
     $(if $(value $1),, \
       $(error Undefined variable $1$(if $2, ($2))))
 
+check-and-create-namespace:
```
Comment (ellistarn) [inline Makefile:92]: I'm a bit confused -- VPCRC should install into `kube-system`. Wondering why your namespace is missing the `kube-` prefix.

File: Makefile [hunk 2]
```diff
@@ -89,7 +89,21 @@ __check_var = \
...
+	else \
+		echo "namespace system exists"; \
+	fi
+
 build-test-binaries:
 	mkdir -p ${MAKEFILE_PATH}build
 	find ${MAKEFILE_PATH} -name '*suite_test.go' -type f  | xargs dirname  | xargs ginkgo build
 	find ${MAKEFILE_PATH} -name "*.test" -print0 | xargs -0 -I {} mv {} ${MAKEFILE_PATH}build
+
+apply-dependencies: check-and-create-namespace
```
Comment (ellistarn) [inline Makefile:107]: Nice!

File: Makefile (hunk 1 above)
Comment (oliviassss) [reply Makefile:92]: @ellistarn, the ns system is for the cert-manager certificate https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/master/config/certmanager/certificate.yaml#L9

File: Makefile [hunk 3]
```diff
@@ -89,7 +89,21 @@ __check_var = \
...
+	bash ${MAKEFILE_PATH}/scripts/test/install-cert-manager.sh
+	kubectl apply -f ${MAKEFILE_PATH}/config/certmanager/certificate.yaml
```
Comment (ellistarn) [inline Makefile:109]: IIUC this should be installed automatically as part of the kustomize command, and thus included in `make apply.
```
❯ kustomize build config/default | grep -C5 Certificate
# Warning: 'bases' is deprecated. Please use 'resources' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
# Warning: 'vars' is deprecated. Please use 'replacements' instead. [EXPERIMENTAL] Run 'kustomize edit fix' to update your Kustomization automatically.
        secret:
          defaultMode: 420
          secretName: webhook-server-cert
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: vpc-resource-serving-cert
  namespace: kube-system
spec:
  dnsNames:
```

Comment (oliviassss) [reply Makefile]: double-checked, after installing the cert-manager, the certificate will be installed during the kustomize cmd.

**Review by ellistarn — DISMISSED** [review]

**Review by ellistarn — APPROVED** [review]
🥳

**Pull request merged**